scipy/optimize/_linprog_util.py

```diff
@@ -51,6 +51,28 @@
         the optimization algorithm. This argument is currently used only by the
         'revised simplex' method, and can only be used if `x0` represents a
         basic feasible solution.
+    integrality : 1-D array or int, optional
+        Indicates the type of integrality constraint on each decision variable.
+
+        ``0`` : Continuous variable; no integrality constraint.
+
+        ``1`` : Integer variable; decision variable must be an integer
+        within `bounds`.
+
+        ``2`` : Semi-continuous variable; decision variable must be within
+        `bounds` or take value ``0``.
+
+        ``3`` : Semi-integer variable; decision variable must be an integer
+        within `bounds` or take value ``0``.
+
+        By default, all variables are continuous.
+
+        For mixed integrality constraints, supply an array of shape `c.shape`.
+        To infer a constraint on each decision variable from shorter inputs,
+        the argument will be broadcasted to `c.shape` using `np.broadcast_to`.
+
+        This argument is currently used only by the ``'highs'`` method and
+        ignored otherwise.
```

Comment:
This is fine and good for consistency, but technically not necessary since `_LPProblem` is private. For this reason, its documentation may be out of date already. When I started working on `linprog` I copy-pasted full docstrings like this in all the private methods, and I regret it a bit because it tends to bloat the files and make it harder to get around. Going forward, I'd prefer to document inputs to private functions only if they are somehow different than they are in the public functions or need notes about how they will be used, and otherwise just refer to the documentation of the public functions.
We don't need to change this because I don't want to re-run CI just for this. Just a thought for the future.